pkg/build/registry/buildconfig/etcd/etcd.go

```diff
@@ -47,3 +51,32 @@ func NewStorage(s storage.Interface) *REST {
 
 	return &REST{store}
 }
+
+func (r *REST) Delete(ctx kapi.Context, name string, options *kapi.DeleteOptions) (runtime.Object, error) {
+	bcObj, err := r.Get(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	bc := bcObj.(*api.BuildConfig)
+
+	// Just update DeletionTimestamp so Builds get deleted
+	if bc.DeletionTimestamp.IsZero() {
+		now := unversioned.Now()
+		bc.DeletionTimestamp = &now
+		bcObj, _, err = r.Update(ctx, bc)
+		return bcObj, err
+	}
+
+	if !bc.Status.CanDelete {
+		err = kapierrors.NewConflict("BuildConfig", fmt.Sprintf("%s/%s", bc.Namespace, bc.Name), fmt.Errorf("The system is ensuring all Builds instantiated from this BuildConfig are removed. Upon completion, this BuildConfig will automatically be purged by the system."))
+		return nil, err
```

Comment:
it's still not an error right?  if i call delete twice on the same bulidconfig, the first time i'm going to get a successful response and the second time i'm going to get an error, which doesn't make sense since there's really no error from an end-user perspective.  the buildconfig is still successfully being deleted just like it was after the first call.